step-functions-plugins/step-functions-plugins-composite/step-functions-plugins-composite-def/src/test/java/step/plugins/functions/types/CompositeResolvedPlanBuilderTest.java

```diff
@@ -0,0 +1,162 @@
+package step.plugins.functions.types;
+
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import step.artefacts.BaseArtefactPlugin;
+import step.artefacts.CallFunction;
+import step.artefacts.Return;
+import step.artefacts.handlers.functions.TokenForcastingExecutionPlugin;
+import step.core.AbstractStepContext;
+import step.core.accessors.AbstractOrganizableObject;
+import step.core.artefacts.reports.aggregated.AggregatedReportView;
+import step.core.artefacts.reports.aggregated.AggregatedReportViewBuilder;
+import step.core.dynamicbeans.DynamicValue;
+import step.core.execution.ExecutionContext;
+import step.core.execution.ExecutionEngine;
+import step.core.execution.ExecutionEngineContext;
+import step.core.plans.Plan;
+import step.core.plans.builder.PlanBuilder;
+import step.core.plans.runner.PlanRunnerResult;
+import step.engine.plugins.AbstractExecutionEnginePlugin;
+import step.engine.plugins.FunctionPlugin;
+import step.functions.Function;
+import step.functions.handler.JsonBasedFunctionHandler;
+import step.functions.io.Input;
+import step.functions.io.Output;
+import step.functions.io.OutputBuilder;
+import step.functions.type.AbstractFunctionType;
+import step.functions.type.FunctionTypeRegistry;
+import step.planbuilder.BaseArtefacts;
+import step.planbuilder.FunctionArtefacts;
+import step.threadpool.ThreadPoolPlugin;
+
+import javax.json.JsonObject;
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+
+import static org.junit.Assert.assertEquals;
+
+public class CompositeResolvedPlanBuilderTest {
+
+    private ExecutionEngine engine;
+
+    @Before
+    public void before() {
+        engine = new ExecutionEngine.Builder().withPlugin(new BaseArtefactPlugin()).withPlugin(new ThreadPoolPlugin())
+                .withPlugin(new FunctionPlugin()).withPlugin(new AbstractExecutionEnginePlugin() {
+                    @Override
+                    public void initializeExecutionContext(ExecutionEngineContext executionEngineContext, ExecutionContext executionContext) {
+                        super.initializeExecutionContext(executionEngineContext, executionContext);
+                        FunctionTypeRegistry functionTypeRegistry = executionContext.require(FunctionTypeRegistry.class);
+                        functionTypeRegistry.registerFunctionType(new CompositeFunctionType(null));
+                        functionTypeRegistry.registerFunctionType(new CustomFunctionType());
+                    }
+                }).withPlugin(new TokenForcastingExecutionPlugin()).build();
+    }
+
+    @After
+    public void after() {
+        engine.close();
+    }
+
+
+    @Test
+    public void planWithCallComposite() throws IOException, InterruptedException {
```

Comment:
We might also cover the case where Keywords are called by dynamic name (i.e. the case where the resolution of the Keyword needs the evaluation of an expression).